src/helpers/metadataUtils.ts

```diff
@@ -25,6 +25,23 @@ export async function decryptNodeData(eciesData: EciesHex, ciphertextHex: string
   return decryptedSigBuffer;
 }
 
+export async function decryptNodeDataWithPadding(eciesData: EciesHex, ciphertextHex: string, privKey: Buffer): Promise<Buffer> {
+  const ciphertextHexPadding = ciphertextHex.padStart(64, "0");
+  const metadata = encParamsHexToBuf(eciesData);
+
+  try {
+    const decryptedSigBuffer = await decrypt(privKey, {
+      ...metadata,
+      ciphertext: Buffer.from(ciphertextHexPadding, "hex"),
+    });
+    return decryptedSigBuffer;
+  } catch (error) {
+    log.warn("Failed to decrypt padded share cipher", error);
+    // try without cipher text padding
+    return decrypt(privKey, { ...metadata, ciphertext: Buffer.from(ciphertextHex, "hex") });
+  }
+}
+
```

Comment:
We shouldn't need this padding. This will always lead to an error. Unless the ciphertext sender removed leading zeros from the ciphertext.

In any case, I think we should try unpadded first.

Also, I think we can keep the name `decryptNodeData`.

Reply:
fixed it to do padding on error, i changed the function name because decryptNodeData is being used at other places as well where padding is not being used.